### PR TITLE
[GEOT-7075] Perform the same tests found in PostgisJSONOnlineTest on JSONB columns

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOT-7075](https://osgeo-org.atlassian.net/browse/GEOT-7075) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java [hunk 1]
```diff
@@ -146,6 +146,22 @@ public void testJSONPointerFunction() throws Exception {
             }
             assertEquals(1, size);
         }
+        if (pgJsonTestSetup.supportJsonB) {
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java:149]: All of these seem basically copy/paste with the name of the column changed. Could you factor out a method that takes the name of the column instead?

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java [hunk 2]
```diff
@@ -37,11 +37,21 @@ protected void setUpData() throws Exception {
 
         try (Connection cx = getConnection();
                 Statement st = cx.createStatement();
-                ResultSet rs = st.executeQuery("select Version()")) {
+                ResultSet rs = st.executeQuery("SELECT current_setting('server_version_num')")) {
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java:40]: Was the old code not working on your setup? Just wondering about this change.

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java [hunk 3]
```diff
@@ -37,11 +37,21 @@ protected void setUpData() throws Exception {
...
+                    supportJsonB = Integer.parseInt(rs.getString(1)) >= 90004;
+                } catch (NumberFormatException nfe) {
+                    // unable to determine PostgreSQL version because non-integer returned
+                    supportJsonB = false;
+                }
             }
+        } catch (PSQLException pse) {
+            // server_version_num was added in PostgreSQL 8.2, throws this exception if the
+            // parameter is
+            // not found
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java:53]:
```suggestion
            // parameter is not found
```

**Review by aaime — CHANGES_REQUESTED** [review]

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java [hunk 4]
```diff
@@ -127,54 +127,77 @@ public void testNullEntry() throws Exception {
...
             assertEquals(1, size);
         }
     }
 
     @Test
-    public void testJSONPointerFunctionWithArray() throws Exception {
+    public void testJSONPointerFunction() throws Exception {
         ContentFeatureSource fs = dataStore.getFeatureSource(tname("jsontest"));
         FilterFactory ff = dataStore.getFilterFactory();
+        testPointerEquals(fs, ff,"jsonColumn","/weight",0.001);
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java:154]: I believe a simpler refactor would have been to extract a function that only takes the name of the property, everything else is actually the same, could have stayed in the delegate method. Just thinkng out loud, but I don't see other calls to testPointerEquals.

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java (hunk 1 above)
Comment (turingtestfail) [reply modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java]: done

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java (hunk 2 above)
Comment (turingtestfail) [reply modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonTestSetup.java:40]: Correct, this was failing on my setup.  Selecting version will often return build information text that was not being handled by the GT version comparison tools.  current_setting is the recommended way to get numerical information from Postgres for comparison.

File: modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java (hunk 4 above)
Comment (turingtestfail) [reply modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java]: Agreed and fixed.

**Pull request merged**